diff --git a/security/Kconfig b/security/Kconfig
index bb24477..8ee328d 100644
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -152,6 +152,7 @@ config LSM_MMAP_MIN_ADDR
 source security/selinux/Kconfig
 source security/smack/Kconfig
 source security/tomoyo/Kconfig
+source security/lsmsb/Kconfig
 source security/apparmor/Kconfig
 
 source security/integrity/ima/Kconfig
@@ -177,6 +178,9 @@ choice
 	config DEFAULT_SECURITY_TOMOYO
 		bool "TOMOYO" if SECURITY_TOMOYO=y
 
+	config DEFAULT_SECURITY_LSMSB
+		bool "LSMSB" if SECURITY_LSMSB=y
+
 	config DEFAULT_SECURITY_APPARMOR
 		bool "AppArmor" if SECURITY_APPARMOR=y
 
@@ -190,6 +194,7 @@ config DEFAULT_SECURITY
 	default "selinux" if DEFAULT_SECURITY_SELINUX
 	default "smack" if DEFAULT_SECURITY_SMACK
 	default "tomoyo" if DEFAULT_SECURITY_TOMOYO
+	default "lsmsb" if DEFAULT_SECURITY_LSMSB
 	default "apparmor" if DEFAULT_SECURITY_APPARMOR
 	default "" if DEFAULT_SECURITY_DAC
 
diff --git a/security/Makefile b/security/Makefile
index fa77021..d8ff6ad 100644
--- a/security/Makefile
+++ b/security/Makefile
@@ -6,6 +6,7 @@ obj-$(CONFIG_KEYS)			+= keys/
 subdir-$(CONFIG_SECURITY_SELINUX)	+= selinux
 subdir-$(CONFIG_SECURITY_SMACK)		+= smack
 subdir-$(CONFIG_SECURITY_TOMOYO)        += tomoyo
+subdir-$(CONFIG_SECURITY_LSMSB)         += lsmsb
 subdir-$(CONFIG_SECURITY_APPARMOR)	+= apparmor
 
 # always enable default capabilities
@@ -20,6 +21,7 @@ obj-$(CONFIG_SECURITY_SELINUX)		+= selin
 obj-$(CONFIG_SECURITY_SMACK)		+= smack/built-in.o
 obj-$(CONFIG_AUDIT)			+= lsm_audit.o
 obj-$(CONFIG_SECURITY_TOMOYO)		+= tomoyo/built-in.o
+obj-$(CONFIG_SECURITY_LSMSB)		+= lsmsb/built-in.o
 obj-$(CONFIG_SECURITY_APPARMOR)		+= apparmor/built-in.o
 obj-$(CONFIG_CGROUP_DEVICE)		+= device_cgroup.o
 
diff --git a/security/lsmsb/Kconfig b/security/lsmsb/Kconfig
new file mode 100644
index 0000000..d5a3c74
--- /dev/null
+++ b/security/lsmsb/Kconfig
@@ -0,0 +1,6 @@
+config SECURITY_LSMSB
+	bool "Linux Security Modules Sandbox"
+	default n
+	help
+	  TODO
+
diff --git a/security/lsmsb/Makefile b/security/lsmsb/Makefile
new file mode 100644
index 0000000..cdf2f17
--- /dev/null
+++ b/security/lsmsb/Makefile
@@ -0,0 +1,5 @@
+#
+# Makefile for LSMSB
+#
+
+obj-$(CONFIG_SECURITY_LSMSB) := lsmsb.o path.o
